hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpConnection.java

```diff
@@ -99,27 +98,27 @@ public Channel getChannel() {
     }
 
     @Override
-    public ConnectionType getType() {
+    public String getType() {
         return type;
     }
 
     @Override
-    public void setType(ConnectionType type) {
-        if (this.type != NONE) {
+    public void setType(String type) {
+        if (!this.type.equals(NONE)) {
             return;
         }
 
         this.type = type;
-        if (type == MEMBER) {
+        if (type.equals(MEMBER)) {
```

Comment:
This check will throw an NPE when `type` argument is `null`, while the previous one wouldn't. Is it OK or we need something like `Objects#requireNonNull` here?

Reply:
Adding null check.